pkg/puppet/puppet.go

```diff
@@ -154,6 +154,22 @@ func kubernetesClusterConfig(conf *clusterv1alpha1.ClusterKubernetes, hieraData
 		hieraData.classes = append(hieraData.classes, `prometheus`)
 	}
 
+	if a := conf.APIServer; a != nil && len(a.FeatureGates) > 0 {
+		hieraData.variables = append(hieraData.variables, fmt.Sprintf(`kubernetes::api_feature_gates: ["%s"]`, strings.Join(a.FeatureGates, `", "`)))
```

Comment:
why not use `kubernetes::apiserver::feature_gates` as var and only define the var in the correct class? We don't need that variable in the `apiserver` class or any other subclass of it.

I have the same remark for the other components.